### PR TITLE
Modify RepositoryMock's behavior to fit GitLab's

### PR DESCRIPTION
**Pull request by mogawinch**
Based on this: https://docs.gitlab.com/ee/api/releases/#create-a-release

When a tag with 'tag_name' does not exist and 'ref' exist, a tag with 'tag_name' will be created when the release will be created.

### CODE REVIEW

File: NGitLab.Mock/ReleaseCollection.cs [hunk 1]
```diff
@@ -40,13 +41,21 @@ public override void Add(ReleaseInfo release)
             base.Add(release);
         }
 
-        public ReleaseInfo Add(string tagName, string name, string description, User user)
+        public ReleaseInfo Add(string tagName, string name, string reference, string description, User user)
```
Comment (louis-z) [inline NGitLab.Mock/ReleaseCollection.cs:49]: This is a breaking change. Should we add a method overload instead?

@meziantou , what do you think?

File: NGitLab.Mock/ReleaseCollection.cs [hunk 2]
```diff
@@ -40,13 +41,21 @@ public override void Add(ReleaseInfo release)
...
         {
             if (tagName is null)
                 throw new ArgumentNullException(nameof(tagName));
             if (user is null)
                 throw new ArgumentNullException(nameof(user));
 
+            if (!Project.Repository.GetTags().Any(r => string.Equals(r.FriendlyName, tagName, StringComparison.Ordinal)))
+            {
+                if (string.IsNullOrEmpty(reference))
+                    throw new GitLabBadRequestException();
```
Comment (louis-z) [inline NGitLab.Mock/ReleaseCollection.cs]: Please add a message indicating why the exception is thrown.

**Review by louis-z — CHANGES_REQUESTED** [review]

File: NGitLab.Mock/ReleaseCollection.cs (hunk 1 above)
Comment (jairbubbles) [reply NGitLab.Mock/ReleaseCollection.cs:49]: (on mocks, I'm not sure we care as long as it's source compatible)

Comment (jairbubbles) [reply NGitLab.Mock/ReleaseCollection.cs:49]: (would be nice to add https://www.nuget.org/packages/Microsoft.CodeAnalysis.PublicApiAnalyzers)

File: NGitLab.Mock/ReleaseCollection.cs (hunk 2 above)
Comment (meziantou) [reply NGitLab.Mock/ReleaseCollection.cs]: done

File: NGitLab.Mock/ReleaseCollection.cs (hunk 1 above)
Comment (meziantou) [reply NGitLab.Mock/ReleaseCollection.cs:49]: I've reintroduced the previous method, so there is no breaking change.

@jairbubbles https://github.com/ubisoft/NGitLab/issues/52

**Review by louis-z — APPROVED** [review]

**Pull request merged**